### ISSUE TITLE
Canot rename bucket

### ISSUE THREAD

**Issue by rawkode** [pulumi/pulumi-aws-native#584]
### What happened?

When changing the `bucketName` property, Pulumi tried to rename the bucket which isn't a valid option.

### Steps to reproduce

```typescript
const bucket = new aws.s3.Bucket("bucket", {
  bucketName: `rawkode-pulumi-workshop-2022-${minScale}`,
});
```

Run `pulumi up` and then change `bucketName` and run `pulumi up` again.

### Expected Behavior

Pulumi replaces the bucket

### Actual Behavior

Error

```

Diagnostics:
  pulumi:pulumi:Stack (scratch2-dev):
    error: update failed

  aws-native:s3:Bucket (bucket):
    error: operation error CloudControl: UpdateResource, https response error StatusCode: 400, RequestID: 2d1cf7ce-9d36-4927-b7da-f3171f025349, NotUpdatableException: Invalid patch update: createOnlyProperties [/properties/BucketName] cannot be updated

Outputs:
  - bucketName: "rawkode-pulumi-workshop-2022-2"

Resources:
    1 unchanged

```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by stack72** [comment]
Thanks for the find here @rawkode - we will get this looked at!

**Comment by mjeffryes** [comment]
This should be resolved by #1053; changes to bucketNAme will result in a replace rather than an update.
